espnet2/sds/asr/espnet_asr.py

```diff
@@ -0,0 +1,49 @@
+import os
+
+import librosa
+import numpy as np
+import torch
+from typeguard import typechecked
+
+from espnet2.bin.asr_inference import Speech2Text
+from espnet2.sds.asr.abs_asr import AbsASR
+
+
+class ESPnetASRModel(AbsASR):
+    """ESPnet ASR"""
+
+    @typechecked
+    def __init__(
+        self,
+        tag=(
+            "espnet/"
+            "simpleoier_librispeech_asr_train_asr_conformer7_"
+            "wavlm_large_raw_en_bpe5000_sp"
+        ),
+        device="cuda",
+    ):
+        super().__init__()
+        self.s2t = Speech2Text.from_pretrained(
+            model_tag=tag,
+            device=device,
+            beam_size=1,
+        )
+        self.device = device
+
+    def warmup(self):
+        with torch.no_grad():
+            dummy_input = (
+                torch.randn(
+                    (3000),
+                    dtype=getattr(torch, "float16"),
```

Comment:
default dtype should become an initialization argument, so it may will allow other than float16 (bfloat16 seems more stable) or probably int or qint

Reply:
I’ve added dtype as an initialization argument. Currently, "bfloat16" is not supported with torch.randn, so I’ve set it to "float16" as a default. Since it is now an argument, this can easily be updated in the future.